nemo/collections/asr/parts/utils/diarization_utils.py

```diff
@@ -505,19 +515,36 @@ def make_json_output(self, uniq_id, diar_hyp, word_dict_seq_list, total_riva_dic
                         {'speaker': {'name': None, 'id': prev_speaker}, 'terms': terms_list}
                     )
                     terms_list = []
-                string_out = self.print_time(speaker, start_point, end_point, self.params, previous_string=string_out)
+
+                # remove trailing space in text
+                sentence['text'] = sentence['text'].strip()
+
+                # store last sentence
+                sentences.append(sentence)
+
+                # start construction of a new sentence
+                sentence = {'speaker': speaker, 'start_point': start_point, 'end_point': end_point, 'text': ''}
             else:
-                string_out = self.print_time(
-                    speaker, start_point, end_point, self.params, previous_string=string_out, replace_time=True
-                )
-            stt_sec, end_sec = round(start_point, 2), round(end_point, 2)
+                # correct the ending time
+                sentence['end_point'] = end_point
+
+            stt_sec, end_sec = start_point, end_point
             terms_list.append({'start': stt_sec, 'end': end_sec, 'text': word, 'type': 'WORD'})
-            string_out = self.print_word(string_out, word, self.params)
+
+            # add current work to sentence
+            sentence['text'] += word.strip() + ' '
+
             self.add_json_to_dict(riva_dict, word, stt_sec, end_sec, speaker)
             audacity_label_words.append(self.get_audacity_label(word, stt_sec, end_sec, speaker))
             total_riva_dict[uniq_id] = riva_dict
             prev_speaker = speaker
 
+        # prepare
+        string_out = self.print_sentences(sentences, self.params)
+
+        # add sentences to json array
+        self.add_sentences_to_dict(riva_dict, sentences)
```

Comment:
can you move this statement to `write_and_log` function, so that all writing to dicts and files would be in one place?